ergo/metaculus.py

```diff
@@ -192,28 +201,81 @@ def sample_community(self):
         raise NotImplementedError("This should be implemented by a subclass")
 
     @staticmethod
-    def to_dataframe(questions: List["MetaculusQuestion"]) -> pd.DataFrame:
+    def to_dataframe(
+        questions: List["MetaculusQuestion"], extra_columns: List[str] = []
+    ) -> pd.DataFrame:
         """
         Summarize a list of questions in a dataframe
 
         :param questions: questions to summarize
+        :param extra_columns: optional list of column names as strings
         :return: pandas dataframe summarizing the questions
         """
+
+        # It's natural to pass a string to extra_columns when you just want to add
+        # one column, e.g., ergo.MetaculusQuestion.to_dataframe(qs, extra_columns="change")
+        # Because Python treats strings similarly to lists, this can lead to cryptic errors.
+        # So I'm adding this explicit error to prevent confusion.
+        if type(extra_columns) != list:
+            raise TypeError("extra_columns must be a list")
+
         show_names = any(q.name for q in questions)
         if show_names:
-            columns = ["id", "name", "title", "resolve_time"]
+            columns = ["id", *extra_columns, "name", "title", "resolve_time"]
             data = [
-                [question.id, question.name, question.title, question.resolve_time]
+                [
+                    question.id,
+                    *[question.data[key] for key in extra_columns],
+                    question.name,
+                    question.title,
+                    question.resolve_time,
+                ]
                 for question in questions
             ]
         else:
-            columns = ["id", "title", "resolve_time"]
+            columns = ["id", *extra_columns, "title", "resolve_time"]
             data = [
-                [question.id, question.title, question.resolve_time]
+                [
+                    question.id,
+                    *[question.data[key] for key in extra_columns],
+                    question.title,
+                    question.resolve_time,
+                ]
                 for question in questions
             ]
         return pd.DataFrame(data, columns=columns)
 
+    def get_i_of_community_prediction_before(self, timestamp: int) -> Optional[int]:
```

Comment:
This function signature feels more natural to me:

```py
def get_community_prediction(self, before: datetime = None):
    ...
  

def change_since(self, since: datetime):
    """
    Calculate change in community prediction between the argument and most recent prediction
    :param since: timestamp
    :return: change in community prediction since timestamp
    """
    try:
        old = self.get_community_prediction(before=since, default=None)
    except LookupError:
        return 0
    new = self.get_community_prediction()
    return new - old
```